server/service/src/requisition_line/response_line_stats/response_line_stats.rs

```diff
@@ -89,18 +89,44 @@ pub fn response_store_stats(
             .status(RequisitionStatus::Finalised.not_equal_to()),
     )?;
 
-    let other_requested_quantity = (response_requisition_lines
-        .iter()
-        .fold(0.0, |sum, requisition_line| {
-            sum + requisition_line.requisition_line_row.requested_quantity
-        }))
-        - requisition_line.requisition_line_row.requested_quantity;
+    // Calculate line using approved amount if exists -> if denied or pending return 0 -> if doesn't need approval, use requested quantity
+    let calculate_line_quantity = |line: &RequisitionLine| -> f64 {
+        match line.requisition_row.approval_status {
+            Some(ApprovalStatusType::Approved)
+            | Some(ApprovalStatusType::ApprovedByAnother)
+            | Some(ApprovalStatusType::AutoApproved) => {
+                let quantity = if line.requisition_line_row.approved_quantity > 0.0 {
+                    line.requisition_line_row.approved_quantity
+                } else {
+                    line.requisition_line_row.requested_quantity
+                };
+                quantity
+            }
+            Some(ApprovalStatusType::Denied)
+            | Some(ApprovalStatusType::DeniedByAnother)
+            | Some(ApprovalStatusType::Pending) => 0.0,
+
+            Some(ApprovalStatusType::None) | None => {
+                let quantity = line.requisition_line_row.requested_quantity;
+                quantity
+            }
+        }
+    };
```

Comment:
```suggestion
    let calculate_line_quantity = |line: &RequisitionLine| -> f64 {
        match line.requisition_row.approval_status {
            Some(ApprovalStatusType::Approved)
            | Some(ApprovalStatusType::ApprovedByAnother)
            | Some(ApprovalStatusType::AutoApproved) => {
                if line.requisition_line_row.approved_quantity > 0.0 {
                    line.requisition_line_row.approved_quantity
                } else {
                    line.requisition_line_row.requested_quantity
                }
            }
            Some(ApprovalStatusType::Denied)
            | Some(ApprovalStatusType::DeniedByAnother)
            | Some(ApprovalStatusType::Pending) => 0.0,

            Some(ApprovalStatusType::None) | None => line.requisition_line_row.requested_quantity,
        }
    };
```
Can return the value directly without declaring it (:

Reply:
Can probably hide the match behind store pref as well so it only calculates if store has response_requisition_requires_authorisation, request_requisition_requires_authorisation these prefs